From 50c86728857ac152745ca75f7eef4cd713124286 Mon Sep 17 00:00:00 2001
Message-Id: <50c86728857ac152745ca75f7eef4cd713124286.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: gsarora <gagandeep.s.arora@intel.com>
Date: Sat, 12 Jul 2014 13:56:24 +0530
Subject: [PATCH 289/312] FOR_UPSTREAM [VPG]: drm/i915: Fix to update the
 correct latency for sprite C and D.

The drain latency values for sprite C and D are not calculated taking
pipe B clock into consideration. Adding fix to update the correct
drain latency value.

Issue: GMIN-2875
Change-Id: I2a9722f19b40d19b221e292511bad38a6e78a055
Signed-off-by: Gagandeep S Arora <gagandeep.s.arora@intel.com>
Signed-off-by: Nabendu Maiti <nabendu.bikash.maiti@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index a4c9c7d..4f48b9b 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -1487,7 +1487,7 @@ static bool g4x_compute_srwm(struct drm_device *dev,
 }
 
 static bool vlv_compute_drain_latency(struct drm_device *dev,
-				int plane,
+				int pipe,
 				int *plane_prec_mult,
 				int *plane_dl,
 				int *cursor_prec_mult,
@@ -1501,7 +1501,7 @@ static bool vlv_compute_drain_latency(struct drm_device *dev,
 	int entries;
 	bool latencyprogrammed = false;
 
-	crtc = intel_get_crtc_for_plane(dev, plane);
+	crtc = intel_get_crtc_for_pipe(dev, pipe);
 	if (!intel_crtc_active(crtc))
 		return false;
 
@@ -3164,8 +3164,9 @@ static void valleyview_update_sprite_wm(struct drm_plane *plane,
 		shift = DDL_SPRITEB_SHIFT;
 	}
 
-	if (enabled && vlv_compute_drain_latency(dev, 0, NULL, NULL, NULL, NULL,
-			&sprite_prec_mult, &sprite_dl, pixel_size, enable)) {
+	if (enabled && vlv_compute_drain_latency(dev, intel_plane->pipe,
+				NULL, NULL, NULL, NULL, &sprite_prec_mult,
+				&sprite_dl, pixel_size, enable)) {
 
 		if (intel_plane->plane == 0) {
 			sprite_prec = (sprite_prec_mult ==
-- 
1.7.9.5

